pulp_rpm/app/serializers/repository.py

```diff
@@ -119,6 +120,20 @@ class RpmPublicationSerializer(PublicationSerializer):
         ),
     )
 
+    def validate(self, data):
+        """Validate data."""
+        if (
+            data["metadata_checksum_type"] not in settings.ALLOWED_CONTENT_CHECKSUMS
+            or data["package_checksum_type"] not in settings.ALLOWED_CONTENT_CHECKSUMS
```

Comment:
@pavelpicka , @daviddavis , for `metadata_checksum_type `, it's enough to have just this check I think.
For `package_checksum_type`, that's probably a good preliminary check however I don't think it's enough to ensure that no forbidden checksum has slipped through.
I believe the `package_checksum_type` is a configuration for the publication and it should work in the majority of cases but it's the best effort.

If for whatever reason a configured checksum is unavailable, we'll use original one in the publication. In Pulp 3 we allow publications with mixed checksum types for packages. See the code for choosing the checksum for a package https://github.com/pulp/pulp_rpm/blob/master/pulp_rpm/app/tasks/publishing.py#L411-L420. We probably need to check and choose an allowed checksum type there.

E.g. I downloaded an on_demand repo where only md5 is available.
I'm creating a publication specifying sha256 as a package_checksum_type. My publication will have md5 only since it's the original one and the sha256 is not available for this content.

Reply:
Yes, I assumed that updating the publishing code was going to be part of https://pulp.plan.io/issues/7537 but however the work is split up is fine with me.

Reply:
OK, great. I also don't have preference in how the work is split.
It's helpful to know that the publish part was considered and will be covered here or later. Thanks.

Reply:
CC @ggainey for visibility

Reply:
@daviddavis @goosemania Between them, https://pulp.plan.io/issues/7853 and https://pulp.plan.io/issues/7854 should cause the "sync md5-only repo on_demand" to fail **at sync time** - that's one of the decisions that is documented in https://hackmd.io/d5y1IaW_QaSJ-DsosMDkjg

(note: 7537 has mutated into "test all of these repos when all the work is done" issue. The four issues in the referenced hackmd doc are the remaining code-changes needed (we think))

Reply:
I had assumed that we were still going to evaluate and possibly update the publish code but I have no strong objection if others feel that's unnecessary.

Reply:
"Belt and suspenders" works for me - make it impossible for this to happen, but check anyway because *8someone** will be 'clever' enough to slip past the sync-time check...

Reply:
@ggainey , I tried to give a simple example. Let me try this one:
Are we allowing to change the set of allowed checksums after installation? I think yes.
What are we doing with content which we have only "bad" checksum types for? I think it's undecided, we should have a meeting. But I heard opinions, that we just keep it in the DB as is but not publish, not expose to the user.
Having all those assumptions in mind :), the scenario would be:
 - I have md5-only content with md5 allowed,
 - I disallowed md5.
 - I'm trying to publish this content.

Let me know if I'm off somewhere.

Reply:
That is--or rather was--correct. The latest thinking is that pulpcore will forbid starting if there are checksum values that are not in ALLOWED_CONTENT_CHECKSUMS and a user will have to run a command to remove these checksum values before starting. I don't know if there's an issue filed for this?

Reply:
Thinking through this ^, I suppose at the very least, the publish code will have to handle the fact that perhaps some checksums are null.